examples/rust-simple/Dockerfile

```diff
@@ -0,0 +1,50 @@
+FROM rust:1.25.0 as builder
+RUN useradd -m build
+
+# Rust SDK depends on https://github.com/pingcap/grpc-rs and it requires CMake and Go
+
+## Cmake
+ENV CMAKE_MINOR_VERSION=v3.10 \
+    CMAKE_FULL_VERSION=3.10.3
+RUN mkdir -p /usr/src/cmake \
+  && curl -fSLO https://cmake.org/files/${CMAKE_MINOR_VERSION}/cmake-${CMAKE_FULL_VERSION}.tar.gz \
+  && curl -fSLO https://cmake.org/files/${CMAKE_MINOR_VERSION}/cmake-${CMAKE_FULL_VERSION}-SHA-256.txt \
+  && sha256sum -c cmake-${CMAKE_FULL_VERSION}-SHA-256.txt 2>&1 | grep OK \
+  && tar xf cmake-${CMAKE_FULL_VERSION}.tar.gz -C /usr/src/cmake --strip-components=1 \
+  && rm -f cmake-${CMAKE_FULL_VERSION}.* \
+  && cd /usr/src/cmake \
+  && ./configure && make -j$(nproc) && make install
+
+## Go
```

Comment:
Love the multi-stage builder image - we should do that more.

Out of curiosity, why do we need Go here?

Reply:
> Out of curiosity, why do we need Go here?

This Rust SDK depends [gRPC-RS](https://github.com/pingcap/grpc-rs).
It needs Go for TLS encryption and some authentication mechanism. 
See also https://github.com/pingcap/grpc-rs#prerequisites.